File: apps/frontend/lib/videoType.ts
const VIDEO_TYPE_DOT_COLOR: Record<string, string> = {
  MV: "var(--accent)",
  Lyric: "oklch(74% 0.12 200)",
  "Audio Track": "oklch(74% 0.1 260)",
  Cover: "oklch(72% 0.12 320)",
  "Live Performance": "oklch(72% 0.14 140)",
  General: "oklch(60% 0.01 260)",
};

const FALLBACK_DOT_COLOR = "oklch(60% 0.01 260)";

export function videoTypeDotColor(videoType: string): string {
  return VIDEO_TYPE_DOT_COLOR[videoType] ?? FALLBACK_DOT_COLOR;
}

File: apps/mobile/lib/constants/video_type.dart
import 'package:flutter/material.dart';

import 'theme.dart';

const Map<String, Color> _videoTypeDotColor = {
  'MV': AppColors.accent,
  'Lyric': Color(0xFF4FB8C9),
  'Audio Track': Color(0xFF93A8E0),
  'Cover': Color(0xFFDE8AC9),
  'Live Performance': Color(0xFF6FC98C),
  'General': Color(0xFF8B8E99),
};

const Color _fallbackDotColor = Color(0xFF8B8E99);

Color videoTypeDotColor(String videoType) {
  return _videoTypeDotColor[videoType] ?? _fallbackDotColor;
}
